onnxruntime/core/framework/utils.cc

```diff
@@ -633,12 +635,24 @@ void DumpNodeOutputs(OpKernelContext& context, const Node& node, const SessionSt
             if (!provider) {
               provider = cpu_execution_provider;
             }
+            std::cout << " Provider=" << provider->Type() << "\n";
 
+            const auto data_type = tensor.DataType();
             if (provider == cpu_execution_provider || tensor_location.mem_type == OrtMemTypeCPUOutput) {
-              const auto data_type = tensor.DataType();
               DispatchOnTensorType(data_type, DumpTensor, tensor, shape);
-            } else {
-              std::cout << " is not on CPU. Provider=" << provider->Type() << "\n";
+            } else if (provider == cuda_execution_provider) {
+              // copy tensor from gpu to cpu then dump
+              auto cpu_allocator = cpu_execution_provider->GetAllocator(0, OrtMemTypeDefault);
+              std::unique_ptr<Tensor> cpu_tensor = std::make_unique<Tensor>(data_type,
+                                                                            shape,
+                                                                            cpu_allocator);
+              GPUDataTransfer data_transfer;
+              auto status = data_transfer.CopyTensor(tensor, *cpu_tensor.get(), 0);
```

Comment:
>CopyTensor [](start = 42, length = 10)

please also note that cudamemcpy is blocking, so once it is enabled, it may impact the real execute sequence.

Reply:
It might delay execution but the result shall not be impacted. Since it is for correctness verification, so I think it is fine (Actually, dumping data is the most slow part).